src/services/kpis/kpis.svc.coffee

```diff
@@ -190,6 +197,37 @@ angular
       return unless kpi.element_watched
       kpi.watchables = [kpi.element_watched].concat(kpi.extra_watchables || [])
 
+    # Logic specific to applying newly fetched data to a dhb KPI.
+    # TODO: Redesign all business logic for v1 / v2 dhb & widget KPIs.
```

Comment:
This comment is not really helpful: can you describe a bit more what should be reworked? What part of the current business logic is causing trouble?